docs/pages/repo/docs/handbook/environment-variables.mdx

```diff
@@ -1,54 +1,137 @@
-# Using environment variables
-
-We currently recommend using `dotenv-cli` as the simplest way to bring your environment variables into your development tasks.
-
-We're actively looking forward to improving the developer experience and ergonomics for your environment variables in a future release of Turborepo.
-
-## With local `turbo`
-
-1. Place all of your variables into the root of your monorepo as a `.env` file.
+import { Tabs, Tab } from '../../../../components/Tabs';
+
+# Using Environment Variables
+
+Because environment variables are not captured in source code they're not as easily shared across machines. To set up environment variables for your repository you will need to take advantage of a number of Turborepo features.
+
+## Working with Next.js and other libraries that load environment variables
+
+If your framework automatically loads environment variables from a particular file for you, you must make sure that `turbo` knows where that framework is going to load them from. This shows a baseline configuration for Next.js and Vite, who load their environment files themselves:
+
+<Tabs items={['Next.js', 'Vite']} storageKey="selected-framework">
+  <Tab>
+    ```jsonc
+    {
+      "$schema": "https://turbo.build/schema.json",
+      "globalDotEnv": [".env"],
+      "pipeline": {
+        "build": {
+          "dotEnv": [".env.production.local", ".env.local", ".env.production", ".env"]
+        },
+        "dev": {
+          "dotEnv": [".env.development.local", ".env.local", ".env.development", ".env"]
+        },
+        "test": {
+          "dotEnv": [".env.test.local", ".env.test", ".env"]
+        }
+      }
+    }
+    ```
+  </Tab>
+  <Tab>
+    ```jsonc
+    {
+      "$schema": "https://turbo.build/schema.json",
+      "globalDotEnv": [".env"],
+      "pipeline": {
+        "build": {
+          "dotEnv": [".env.production.local", ".env.production", ".env.local", ".env"]
+        },
+        "dev": {
+          "dotEnv": [".env.development.local", ".env.development", ".env.local", ".env"]
+        },
+        "test": {
+          "dotEnv": [".env.test.local", ".env.test", ".env.local", ".env"]
+        }
+      }
+    }
+    ```
+  </Tab>
+</Tabs>
+
+## Loading Your Own Variables
+
+If you need to load a large number of environment variables into your environment just-prior to execution, we recommend using `dotenv-cli`. It's the simplest way to bring your environment variables into your development tasks.
+
+Turborepo does not load any `.env` files into the environment! Your task must handle loading of the `.env` files itself.
```

Comment:
I would use `<Callout>` here.

Reply:
It's a callout in the other spot I copied it from; since we're doing a walkthrough here it felt like gilding the lily. Will switch to callout.